core/src/test/java/org/elasticsearch/transport/netty/NettyTransportRequestLimitIT.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.transport.netty;
+
+import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.common.network.NetworkModule;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
+import org.elasticsearch.test.ESIntegTestCase.Scope;
+
+import java.net.InetAddress;
+import java.util.Locale;
+
+import static org.elasticsearch.common.settings.Settings.settingsBuilder;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+@ClusterScope(scope = Scope.SUITE, minNumDataNodes = 1, maxNumDataNodes = 5, numClientNodes = 0)
+public class NettyTransportRequestLimitIT extends ESIntegTestCase {
```

Comment:
>  This test is irrelevant (and would also fail) with the node client so I had to prevent it (@ClusterScope.transportClientRatio = 1).

I think a better work around (and maybe even a better test) is to explicitly acquire a  client of one node (doesn't matter which type) and send the request to another node (for example, a node info request for a specific node, or a bulk for a single shard index, where you know where the shard is).


> If we had a possibility to randomize the transport (either local or netty) 

Our CI randomizes the transport implementation based on parameters (I think it was node.local=true/false , but it may have changed). Using it means you don't have to do all the hary stuff like configuring ports and such.